### PR TITLE
Add filters to bulk enrollment

### PR DESCRIPTION
**Pull request by raq929**
This PR

- Refactors pagination
- Adds the SearchHeader from `frontend-enterprise`
- Updates the table to link to enterprises and show the start date of the course
- Upgrades Paragon and the open-edx theme

The merged header/filters has been approved by Jenn.

Screen Shot 2021-02-22 at 5 02 46 PM

Screen Shot 2021-02-23 at 1 14 52 PM

### CODE REVIEW

File: src/components/BulkEnrollmentPage/BulkEnrollement.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.bg-brand-primary {
```
Comment (edx-abolger) [inline src/components/BulkEnrollmentPage/BulkEnrollement.scss:1]: Can we rename for correct spelling?

File: src/components/BulkEnrollmentPage/ConnectedPagination.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  DataTable, DataTableContext,
+} from '@edx/paragon';
+
+export const BaseConnectedPagination = ({
+  currentRefinement: algoliaPageNo,
+  refine,
+}) => {
+  const { state: tableState } = useContext(DataTableContext);
+  useEffect(() => {
+    if (algoliaPageNo && tableState.pageIndex + 1 !== algoliaPageNo) {
```
Comment (edx-abolger) [inline src/components/BulkEnrollmentPage/ConnectedPagination.jsx:16]: nit: is it against style guidelines to use an extra set of parentheses here?

File: src/components/BulkEnrollmentPage/ConnectedPagination.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      // algolia is 1 indexed and react-table is 0 indexed
```
Comment (edx-abolger) [inline src/components/BulkEnrollmentPage/ConnectedPagination.jsx:17]: To discuss: docs imply that Algolia is _not_ 1-indexed. https://www.algolia.com/doc/api-reference/api-parameters/page/#usage-notes

File: src/components/BulkEnrollmentPage/ConnectedPagination.test.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Comment (edx-abolger) [inline src/components/BulkEnrollmentPage/ConnectedPagination.test.jsx:1]: Potentially relevant: is there no test related to the functionality of the hitsPerPage/ page Size values?

**Review by edx-abolger — CHANGES_REQUESTED** [review]
Want to talk about pagination thing with algolia

File: src/components/BulkEnrollmentPage/BulkEnrollement.scss (hunk 1 above)
Comment (raq929) [reply src/components/BulkEnrollmentPage/BulkEnrollement.scss:1]: I can't spell enrollment or license to save my life. Will fix.

File: src/components/BulkEnrollmentPage/ConnectedPagination.jsx (hunk 2 above)
Comment (raq929) [reply src/components/BulkEnrollmentPage/ConnectedPagination.jsx:16]: No, it's not, and I think we should.

File: src/components/BulkEnrollmentPage/ConnectedPagination.test.jsx (hunk 4 above)
Comment (raq929) [reply src/components/BulkEnrollmentPage/ConnectedPagination.test.jsx:1]: This component doesn't deal with hits per page or page size at all. All it does is call algolia if the table's page number is different from Algolia's page number.
Worth checking if my tests on the SearchResults component have tests for those things, tho.

File: src/components/BulkEnrollmentPage/CourseSearchResults.test.jsx [hunk 5]
```diff
@@ -40,21 +51,44 @@ const CourseSearchWrapper = props => (
...
+
+describe('<FormattedDateCell />', () => {
+  it('renders a formatted date', () => {
+    const wrapper = mount(<FormattedDateCell value={testStartDate} />);
+    expect(wrapper.text()).toEqual('Sep 10, 2020');
+  });
+});
+
+describe('<CourseSearchResults />', () => {
   it('renders search results', () => {
```
Comment (raq929) [inline src/components/BulkEnrollmentPage/CourseSearchResults.test.jsx:78]: Add a test to show we are passing the correct page params to the DataTable.

File: src/components/BulkEnrollmentPage/ConnectedPagination.jsx (hunk 3 above)
Comment (raq929) [reply src/components/BulkEnrollmentPage/ConnectedPagination.jsx]: I'm no longer using this component.

File: src/index.scss [hunk 6]
```diff
@@ -5,6 +5,7 @@ $modal-max-width: 650px;
 @import "~@edx/brand/paragon/variables";
 @import "~@edx/paragon/scss/core/core";
 @import "~@edx/brand/paragon/overrides";
+@import "~@edx/frontend-enterprise/dist/frontend-enterprise";
```
Comment (binodpant) [inline src/index.scss:8]: interesting, this import pattern, would have been nice to get something like `~@edx/frontend-enterprise/styles` but okay...

**Comment by binodpant** [comment]
overall looking good: 

@raq929 one question:

I noted in our ticket we had noted that we will only allow filtering by these fields for MVP:

```
allow filtering for these fields only:

-Course name 

-Partner name 

-Subject
```

however in screenshot I see: course, program, partner level, availability

Did we change this decision recently?

File: src/index.scss (hunk 6 above)
Comment (raq929) [reply src/index.scss:8]: Yeah, I don't love it, tbh, but we have a ticket to update the build to Webpack and hopefully we can change it then.

**Comment by raq929** [comment]
@binodpant Yeah, there ended up being technical difficulties using the filters from Algolia, so I ended up using the same SearchHeader as LearnerPortal, and it isn't configurable.

File: package.json [hunk 7]
```diff
@@ -20,9 +20,10 @@
   "license": "AGPL-3.0",
   "dependencies": {
     "@babel/plugin-transform-runtime": "7.12.1",
-    "@edx/brand": "npm:@edx/brand-openedx@1.1.0",
+    "@edx/brand": "npm:@edx/brand-edx.org@^1.4.2",
```
[the remark this answers is not in the document]
Comment (raq929) [reply package.json:23]: Yeah, I think that's an interesting idea. I know people have had trouble with that module.config file at times, though.

**Review by edx-abolger — APPROVED** [review]

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (binodpant) [reply package.json:23]: can the module federation tech be useful here too? I am thinking just injecting various things (themes, local deps etc) dynamically Dependency injection style. can evertything overridable be based on this tech?

**Review by binodpant — APPROVED** [review]
lgtm! yeah we can discuss the filters later does not seem to be an issue to me if we can filte by more fields than anticipated unless we want to actively block those fields will circle back in ux meeting

**Pull request merged**